src/posts/comparing-methods-for-accessing-secrets-in-vault-from-kubernetes.md

```diff
@@ -0,0 +1,285 @@
+
+|  | Secrets Engines | Auth Methods | Sync to K8s Secrets | Secret Rotation |
+| --- | --- | --- | --- | --- |
+| External Secrets Operator | KV only | Token, AppRole, Kubernetes, LDAP, JWT/OIDC | Yes | Yes, if mounted as file |
+
+### ArgoCD Vault Plugin
+
+The [ArgoCD Vault Plugin](https://argocd-vault-plugin.readthedocs.io/en/stable/) requires that you use [ArgoCD](https://argo-cd.readthedocs.io/en/stable/) for managing your Kubernetes deployments. It is the only method in our list that has any kind of dependency on tooling (other than Vault itself), but it is quite an elegant approach and ArgoCD is very popular so we figured it should be included.
+
+The plugin works by templating placeholders in Kubernetes manifests with secrets read from secrets managers (like [HashiCorp Vault](https://argocd-vault-plugin.readthedocs.io/en/stable/backends/#hashicorp-vault), yet [others are supported](https://argocd-vault-plugin.readthedocs.io/en/stable/backends/) also). Obviously you do not want to put your sensitive values straight into your Kubernetes objects that are not made for storing sensitive values, so likely you sensitive values are templated into Kubernetes secrets. However, this does mean if you store non-sensitive values in Vault (e.g. cloud resource IDs, or other configurations), then you **could template those** directly into your Kubernetes manifests. This is a fairly opinionated topic, so let’s say that "it depends" whether that’s a good thing or not. But it’s an interesting side effect that solves many common problems with passing values from infrastructure tools like Terraform to your Kubernetes manifests.
+
+![k8s-argocd-vault-plugin](/blogs/comparing-methods-for-accessing-secrets-in-vault-from-kubernetes/k8s-argocd-vault-plugin.png)
+
+#### Effort
+
+**Platform Effort**: (Assuming you already have ArgoCD) This requires [installing the plugin](https://argocd-vault-plugin.readthedocs.io/en/stable/installation/) for which the [traditional approach is deprecated](https://argo-cd.readthedocs.io/en/stable/operator-manual/config-management-plugins/#configmap-plugin) and the newer [sidecar-based approach](https://argo-cd.readthedocs.io/en/stable/operator-manual/config-management-plugins/#sidecar-plugin) is recommended, and much more maintainable.
+
+**Deployment Effort**: Writing the manifests can be a bit clumsy to get right but the ArgoCD UI makes it quite clear when something did not work, which helps a lot. And because you are templating raw Kubernetes manifests, it is also very easy to check that everything went as intended.
```

Comment:
```suggestion
**Deployment Effort**: Writing the manifests can be a bit clumsy to get right but the ArgoCD UI makes it quite clear when something did not work, which helps a lot. Additionally, because you are templating raw Kubernetes manifests, it is also very easy to check that everything went as intended.
```